Sources/MapboxNavigation/RouteLineController.swift

```diff
@@ -138,7 +138,11 @@ extension NavigationMapView {
             }
         }
         
-        internal func locationUpdate(newLocation: Location) {
+        func locationUpdate(newLocation: Location) {
+            // Since `navigationViewData` is stored weakly in rare cases it might be deallocated before
+            // running current method. Such check prevents crash in such situations.
+            if navigationViewData == nil { return }
```

Comment:
Could there be a race condition in which `navigationViewData` goes away after this check but before the code that’s crashing?

Reply:
Based on tests I do not see that it can occur, `locationUpdate(newLocation:)` is always called from the main thread.